opm/simulators/wells/StandardWellConnections.cpp

```diff
@@ -33,13 +33,125 @@
+                                  const typename std::vector<Scalar>::size_type perf,
+                                  const Properties&                             props,
+                                  const std::vector<Scalar>&                    previousMixture,
+                                  std::vector<Scalar>&                          currentMixture,
+                                  Opm::DeferredLogger&                          deferred_logger)
+    {
+        const auto threshold = static_cast<Scalar>(1.0e-12);
+
+        Scalar rs {0};
+        if (!props.rsmax_perf.empty() && (previousMixture[oilpos] > threshold)) {
+            rs = std::min(previousMixture[gaspos] / previousMixture[oilpos], props.rsmax_perf[perf]);
+        }
+
+        Scalar rv {0};
+        if (!props.rvmax_perf.empty() && (previousMixture[gaspos] > threshold)) {
+            rv = std::min(previousMixture[oilpos] / previousMixture[gaspos], props.rvmax_perf[perf]);
+        }
+
+        const Scalar denom = Scalar{1} - rs*rv;
+        if (! (denom > Scalar{0})) {
+            const auto msg = fmt::format(R"(Problematic denominator value {} for well {}.
+    Connection density calculation with (Rs, Rv) = ({}, {}).
+    Proceeding as if no dissolution or vaporisation for this connection)",
+                                         denom, well, rs, rv);
+
+            deferred_logger.debug(msg);
+        }
+        else {
+            if (rs > Scalar{0}) {
+                // Subtract gas in oil from gas mixture
+                currentMixture[gaspos] =
+                    (previousMixture[gaspos] - previousMixture[oilpos]*rs) / denom;
+            }
+
+            if (rv > Scalar{0}) {
+                // Subtract oil in gas from oil mixture
+                currentMixture[oilpos] =
+                    (previousMixture[oilpos] - previousMixture[gaspos]*rv) / denom;
+            }
+        }
+    }
+
+    template <typename Scalar, typename Properties>
+    void reapportionGasWaterMixture(std::string_view                              well,
+                                    const typename std::vector<Scalar>::size_type gaspos,
+                                    const typename std::vector<Scalar>::size_type waterpos,
+                                    const typename std::vector<Scalar>::size_type perf,
+                                    const Properties&                             props,
+                                    const std::vector<Scalar>&                    previousMixture,
+                                    std::vector<Scalar>&                          currentMixture,
+                                    Opm::DeferredLogger&                          deferred_logger)
+    {
+        // matrix system: (mix[oilpos] = q_os, x[oilpos] = bo*q_or, etc...)
+        // ┌             ┐   ┌                ┐  ┌           ┐
+        // │mix[oilpos]  │   | 1     Rv     0 |  |x[oilpos]  |
```

Comment:
This comment is outdated with new names for the variables. Also, it applies to both this and the above function, maybe create an explanatory comment above both, explaining them in light of this equation?

Reply:
I've pushed an update to move this comment block above `reapportion*Mixture()` helper functions, including to switch to the `componentMixture` and `phaseMixture` naming.